server/hotrod/src/main/resources/META-INF/services/org.infinispan.factories.components.ModuleMetadataFileFinder

```diff
@@ -0,0 +1 @@
+org.infinispan.server.hotrod.ServerMetadataFileFinder
```

Comment:
Ok, seems like other components are being worked on to improve the API to get segments. I'm happy with that.

Just one minor thing, I'm not very keen on having segments be `null` to indicate that all segments should be iterated over. I'd rather use the empty set as way to signal that and don't allow `null`. In this case, that's better IMO than adding Optional in case anyone's wondering. Besides, if an empty set is passed, the javadoc is not clear about what happens.